### ISSUE TITLE
`IncrementSet` should accept a `long`

### ISSUE THREAD

**Issue by hognevevle** [algolia/algoliasearch-client-csharp#775]
Hi,

In the [docs](https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/in-depth/handling-concurrency-with-versioning/) it's suggested to use `IncrementSet` together with a Unix timestamp. 

The [IncrementSet operation in the C# SDK](https://github.com/algolia/algoliasearch-client-csharp/blob/e37d0a8d7e7c6ca051ab1c2a4e49c522ab899220/src/Algolia.Search/Models/Search/PartialUpdateOperation.cs#L75), however, only accepts an `int`, while `long` should be used for unix timestamps. 

I'm using `DateTimeOffset.ToUnixTimeSeconds()` , which returns a `long`. Converting this to an `int` would be possible (at least for a few years), but it still prevents me from using `.ToUnixTimeMilliseconds()` which I would prefer.

Any way you could allow a long here?

**Comment by TomKlotzPro** [comment]
Hey @hognevevle, 

Thanks for this feedback, I'll check with the engine-team if we can send a long. 

I'll keep you updated about this.

**Comment by TomKlotzPro** [comment]
Hello @hognevevle,

I've checked with the engine team, and indeed, this should be a long! Do you want to open the PR to update the code ? I could do it but next week. 😞 

However, I'm preparing a new minor release for next week, so it fits really well.

**Comment by hognevevle** [comment]
Opened #777 . Decided to add new methods instead of updating existing ones, to avoid a breaking change.

**Comment by TomKlotzPro** [comment]
Thank you 🙇 
I'll take a look today, nice job 👍

**Comment by sbellone** [comment]
Hi @hognevevle, thanks for the PR!

> but it still prevents me from using .ToUnixTimeMilliseconds() which I would prefer

Just note that you can still build the operation yourself:
```csharp
var record = new Record
{
    ObjectID = "myID",
    Count = new PartialUpdateOperation<long>
    {
        Operation = PartialUpdateOperationType.IncrementFrom,
        Value = (long)2
    },
};
```

But it's indeed handy to have it built-in 👍

**Comment by sbellone** [comment]
Merged through #799, thanks again! 👍
